fn/list.go

```diff
@@ -0,0 +1,279 @@
+// Copyright (c) 2009 The Go Authors. All rights reserved.
+// Copyright (c) 2024 Lightning Labs and the Lightning Network Developers
+
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+
+//    * Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//    * Redistributions in binary form must reproduce the above
+// copyright notice, this list of conditions and the following disclaimer
+// in the documentation and/or other materials provided with the
+// distribution.
+//    * Neither the name of Google Inc. nor the names of its
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
```

Comment:
Q: what is this license stuff, or why here but not in other places?

Reply:
I source ripped the [golang std List](https://cs.opensource.google/go/go/+/refs/tags/go1.22.4:src/container/list/list.go) and the [BSD License](https://cs.opensource.google/go/go/+/refs/tags/go1.22.4:LICENSE) they include stipulates that we have to reproduce the license.